pkg/middleware/payment/middleware_test.go

```diff
@@ -0,0 +1,307 @@
+package payment
```

Comment:
consider using `payment_test` package, to discourage yourselves from accessing private things in tests.